pkg/provisioner/reconciler/claim-reconciler/helpers.go

```diff
@@ -38,12 +38,24 @@ func claimRefForKey(key client.ObjectKey, ic *internalClient) (types.UID, error)
 	return claim.UID, nil
 }
 
-// Return true if this storage class was for a new bkt vs an existing bkt.
-// referenced storage class.
-func scForNewBkt(sc *storagev1.StorageClass) bool {
+// Return true if this storage class is for a new bucket vs an existing bucket.
+func isNewBucketByClass(sc *storagev1.StorageClass) bool {
 	return len(sc.Parameters[v1alpha1.StorageClassBucket]) == 0
 }
 
+// Return true if this OB is for a new bucket vs an existing bucket.
+func (ic *internalClient) isNewBucketByOB(ob *v1alpha1.ObjectBucket) bool {
```

Comment:
the internalClient should be a shallow encapsulation of a client.Client coupled with a context the controller's scheme.  This should be a helper that takes a `client.Client` interface and then be passed an internalClient.